src/plugins/Workers/TransactionSyncStreamWorker/methods/syncUpToTheGapLimit.js

```diff
@@ -64,15 +65,33 @@ module.exports = async function syncUpToTheGapLimit({
           for (const transaction of walletTransactions.transactions) {
             // eslint-disable-next-line no-underscore-dangle
             const transactionHash = transaction._getHash().reverse().toString('hex');
+
+            self.pendingRequest[transactionHash] = { isProcessing: true, type: 'transaction' };
             // eslint-disable-next-line no-await-in-loop
             const getTransactionResponse = await this.transport.getTransaction(transactionHash);
+
+            if (!getTransactionResponse.blockHash) {
+              // TODO: We should set-up a retry of fetching the tx and it's blockhash
+            }
+            if (getTransactionResponse.blockHash) {
+              self.pendingRequest[getTransactionResponse.blockHash.toString('hex')] = { isProcessing: true, type: 'blockheader' };
+              // eslint-disable-next-line no-await-in-loop
+              const getBlockHeaderResponse = await this
```

Comment:
For browser, as there is an async/await, it will be optimized for that into es5.
is there an issue I do forget here ? 
To explain, we require on reception of the sync works, to fetch the corresponding blockheader of the transaction, which might not even yet exist (as it might not yet be included in a block).
Therefore we set that in pending request so that awaiting for onStop() will properly wait for this to happen before stopping (and therefore avoid having incomplete UTXO set onStart)

Reply:
Yeah, basically when you use `await` inside `for` look it will require babel to include regenerator which is huge. ESlinter usually warns about that.

Reply:
This can indeed be optimized ! Will look at it

Reply:
Do we have `await` in `for` loop somewhere else in Wallet. If so we can keep and optimize it in a separate PR

Reply:
Good idea. 
We do, so a next PR to fix that all seems an awesome idea.